### PR TITLE
*: set MAX GOMAXPROCS in test as 16

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by xhebox — APPROVED** [review]

File: testkit/testenv/testenv.go [hunk 1]
```diff
@@ -22,5 +22,5 @@ import (
 
 // SetGOMAXPROCSForTest sets GOMAXPROCS to 8 if it is greater than 8.
```
Comment (Defined2014) [inline testkit/testenv/testenv.go:23]: change the comments.

Comment (Defined2014) [inline testkit/testenv/testenv.go:23]:
```suggestion
// SetGOMAXPROCSForTest sets GOMAXPROCS to 16 if it is greater than 16.
```

Comment (hawkingrei) [reply testkit/testenv/testenv.go]: Thank you.

**Pull request merged**